react/RuiExpandCollapseReact/README.md

```diff
@@ -0,0 +1,54 @@
+# Expand Collapse
+
+## Internally controlled
+If you do not need to implement any custom logic, then you just need to provide the component an element for the summary content and the details content that is revealed on expand/collapse. Clicking the summary content will trigger an expand/collapse.
+
+```html preview
+<RuiExpandCollapse>
+    <p slot="summary-content">Expander summary</p>
+    <p slot="details-content">Expanded content</p>
+</RuiExpandCollapse>
+```
+
+If you wish for the expand collapse to default to open, use the below:
+```html preview
+<RuiExpandCollapse open={true}>
+    <p slot="summary-content">Expander summary</p>
```

Comment:
Please don't let me derail this PR, just wanted to throw my two cents in.

The API for this component is heavily influenced by the technology it's implemented with, which makes sense now but will get weird in the future.

To me this `slot="summary-content"` is implementation details and the user has to know what each slot name is, if we weren't using web components we'd be using props or child components instead.

I'd suggest either:
```jsx
<RuiExpandCollapse 
    open={true}
    summaryContent={'Expander summary'}
    detailsContent={<p>Expanded content</p>}
/>
```
or
```jsx
<RuiExpandCollapse open={true}>
    <RuiExpandCollapse.Summary>Expander summary</RuiExpandCollapse.Summary>
    <RuiExpandCollapse.Details>
       <p>Expanded content</p>
    </RuiExpandCollapse.Details>
</RuiExpandCollapse>
```

Reply:
That's a really good point - we certainly want to make the dev experience as smooth as possible in each of the technologies and they should be shielded from the specifics of web components. I like the look of the bottom example, @jennasalau what do you think?

Reply:
@jeffdowdle @jennasalau  I have updated the code to implement Jeff's bottom example API. The summary element renders a span by default and the details element renders a div by default. Both can be overriden via an `as` prop if you need to render something else as the root el.

Reply:
> To me `slot="summary-content"` is implementation details and the user has to know what each slot name is,

I agree with this actually... seems like having a slot name means there is a lot of room for errors in the future when there are a large number of components to keep track of

Reply:
I'll have a look at the updated code, i like that we are having discussions like this now. 

Ideally the API level of our component is just data; and slots is one way of passing data in. Attributes is another. The goal to me is hopefully we encapsulate all the display/layout stuff inside the web component.